pkg/k8shandler/certificates.go

```diff
@@ -2,163 +2,111 @@ package k8shandler
-		"tls.crt":       "system.logging.fluentd.crt",
-	},
+var mutex sync.Mutex
+
+//Syncronize blocks single threads access using the certificate mutex
+func Syncronize(action func() error) error {
+	mutex.Lock()
+	defer mutex.Unlock()
+	return action()
 }
 
-func (clusterRequest *ClusterLoggingRequest) extractSecretToFile(secretName string, key string, toFile string) (err error) {
-	secret, err := clusterRequest.GetSecret(secretName)
+func (clusterRequest *ClusterLoggingRequest) extractMasterCerts() (err error) {
+	secret, err := clusterRequest.GetSecret(constants.MasterCASecretName)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			return err
+			return nil
 		}
-		return fmt.Errorf("Unable to extract secret %s to file: %v", secretName, err)
+		return fmt.Errorf("Unable to get secret %s: %v", constants.MasterCASecretName, err)
 	}
-
-	value, ok := secret.Data[key]
-
-	// check to see if the map value exists
-	if !ok {
-		if deprecatedKeys.Has(key) {
-			log.Info("No secret data found. Please be aware but likely not an issue for deprecated keys", "key", key, "deprecatedKeys", deprecatedKeys.List())
-		} else {
-			log.Info("No secret data found", key)
-
+	workDir := utils.GetWorkingDir()
+	for name, value := range secret.Data {
+		if err != utils.WriteToWorkingDirFile(path.Join(workDir, name), value) {
+			return err
 		}
-		return nil
 	}
 
-	return utils.WriteToWorkingDirFile(toFile, value)
+	return nil
 }
 
 func (clusterRequest *ClusterLoggingRequest) writeSecret() (err error) {
 
+	secrets, err := loadFilesFromWorkingDir()
+	if err != nil {
+		return err
+	}
 	secret := NewSecret(
 		constants.MasterCASecretName,
 		clusterRequest.Cluster.Namespace,
-		map[string][]byte{
-			"masterca":  utils.GetWorkingDirFileContents("ca.crt"),
-			"masterkey": utils.GetWorkingDirFileContents("ca.key"),
-		})
-
+		secrets,
+	)
 	utils.AddOwnerRefToObject(secret, utils.AsOwner(clusterRequest.Cluster))
 
-	err = clusterRequest.CreateOrUpdateSecret(secret)
-	if err != nil {
-		return
-	}
-
-	return nil
+	return clusterRequest.CreateOrUpdateSecret(secret)
 }
 
-func (clusterRequest *ClusterLoggingRequest) readSecrets() (err error) {
-
-	for secretName, certMap := range secretCertificates {
-		if err = clusterRequest.extractCertificates(secretName, certMap); err != nil {
-			return
-		}
+func loadFilesFromWorkingDir() (map[string][]byte, error) {
+	workDir := utils.GetWorkingDir()
+	files, err := ioutil.ReadDir(workDir)
+	if err != nil {
+		return nil, err
 	}
-
-	return nil
-}
-
-func (clusterRequest *ClusterLoggingRequest) extractCertificates(secretName string, certs map[string]string) (err error) {
-
-	for secretKey, certPath := range certs {
-		if err = clusterRequest.extractSecretToFile(secretName, secretKey, certPath); err != nil {
-			if errors.IsNotFound(err) {
-				return nil
-			}
-			return
-		}
+	results := map[string][]byte{}
+	for _, f := range files {
+		results[f.Name()] = utils.GetFileContents(path.Join(workDir, f.Name()))
```

Comment:
We could use a check for GetFileContents() failing here.

Reply:
The called function logs an error and returns nil.  Adding check for nil and log message. Skipping adding the file if the content is nil